### PR TITLE
Update PostgreSQL ebuilds

### LINKED ISSUE #532: Update postgresql ebuild to support new Pythons
The current ebuild:

```
PYTHON_COMPAT=( python2_7 python3_{5,6} )
```

Suggestion:
```
PYTHON_COMPAT=( python2_7 python3_{5,6,7,8,9} )
```

### PR DESCRIPTION
**Pull request by xmhd**
### Don't merge this until multiple people have looked at it.

- Ebuilds have been updated to the latest from upstream (Gentoo).
- Ebuilds now have the latest PYTHON_TARGETS (fixes #532).
- Ebuilds are now all EAPI 7.
- PostgreSQL minor versions are all up to date.
- USE=bagger and index.patch has been re-added.
- USE=ltree and wo-ltree.patch has been re-added.
- USE=cassert has been re-added for PostgreSQL 9,
- USE=threads has been set as default for all PostgreSQL ebuilds.
- KEYWORDS has been changed to amd64 only.
- All ebuilds have been revision bumped by 1 revision to reflect our changes.

### Issues

PostgreSQL 10 currently doesn't build against ICU-68. PostgreSQL don't plan on fixing it afaik, so I'll need to find a patch or knock one together, before submitting it to Gentoo. It will build fine with USE="-icu", though.

These ebuilds have all been tested locally on a Gentoo Linux ~amd64 machine with various USE flag configurations (bagger && ltree) and all build successfully.

### CODE REVIEW

File: dev-db/postgresql/postgresql-10.15.ebuild [hunk 1]
```diff
@@ -110,6 +110,12 @@ src_prepare() {
 			die 'PGSQL_PAM_SERVICE rename failed.'
 	fi
 
+        # bagger
+        if use bagger ; then
+                eapply "${FILESDIR}"/${PN}-10.0-index.patch || \
+                        die "failed to apply bagger index patch"
```
Comment (monsieurp) [inline dev-db/postgresql/postgresql-10.15.ebuild]: eapply dies automatically so you can remove the die call here.

edit: remove it everywhere actually :)

File: dev-db/postgresql/postgresql-10.15.ebuild [hunk 2]
```diff
@@ -90,6 +90,14 @@ pkg_setup() {
 }
 
 src_prepare() {
+
+        # make ltree optional
+        eapply "${FILESDIR}/${PN}-10-wo-ltree.patch" || die "failed ltree patch"
```
Comment (monsieurp) [inline dev-db/postgresql/postgresql-10.15.ebuild]: ditto

**Review by monsieurp — CHANGES_REQUESTED** [review]

File: dev-db/postgresql/postgresql-10.15.ebuild (hunk 2 above)
Comment (xmhd) [reply dev-db/postgresql/postgresql-10.15.ebuild]: Good catch  - I have amended the ebuilds.

File: dev-db/postgresql/postgresql-10.15.ebuild (hunk 1 above)
Comment (xmhd) [reply dev-db/postgresql/postgresql-10.15.ebuild]: Amended.

**Review by monsieurp — APPROVED** [review]

**Comment by xmhd** [comment]
All PostgreSQL packages bumped with security fixes for CVE-2021-3393 and CVE-2021-20229.

Also includes a fix for the PostgreSQL 10 build failure with USE="icu".

**Review by alip — COMMENTED** [review]
Tested locally on a ~amd64 Gentoo Linux box with various USE flag combinations, no build failures. However I noticed some minor issues which are listed below:

### ltree

The ltree patch does not completely disable the installation of ltree related files on 9.6, 11, 12 and 13. On 9.6 the effect is minor as it only pertains to the ltree python module. The patch for version 10 works correctly. See the list of installed files below for the whole picture. Relevant flags are `USE=-ltree python`:

```
% qlist postgresql | grep ltree | sort -t- -k2n
/usr/lib64/postgresql-9.6/lib64/ltree_plpython3.so
/usr/share/postgresql-9.6/extension/ltree_plpython2u--1.0.sql
/usr/share/postgresql-9.6/extension/ltree_plpython2u.control
/usr/share/postgresql-9.6/extension/ltree_plpython3u--1.0.sql
/usr/share/postgresql-9.6/extension/ltree_plpython3u.control
/usr/share/postgresql-9.6/extension/ltree_plpythonu--1.0.sql
/usr/share/postgresql-9.6/extension/ltree_plpythonu.control
/usr/include/postgresql-11/server/extension/ltree/ltree.h
/usr/lib64/postgresql-11/lib64/ltree_plpython3.so
/usr/lib64/postgresql-11/lib64/ltree.so
/usr/share/postgresql-11/extension/ltree--1.0--1.1.sql
/usr/share/postgresql-11/extension/ltree--1.1.sql
/usr/share/postgresql-11/extension/ltree.control
/usr/share/postgresql-11/extension/ltree_plpython2u--1.0.sql
/usr/share/postgresql-11/extension/ltree_plpython2u.control
/usr/share/postgresql-11/extension/ltree_plpython3u--1.0.sql
/usr/share/postgresql-11/extension/ltree_plpython3u.control
/usr/share/postgresql-11/extension/ltree_plpythonu--1.0.sql
/usr/share/postgresql-11/extension/ltree_plpythonu.control
/usr/share/postgresql-11/extension/ltree--unpackaged--1.0.sql
/usr/include/postgresql-12/server/extension/ltree/ltree.h
/usr/lib64/postgresql-12/lib64/ltree_plpython3.so
/usr/lib64/postgresql-12/lib64/ltree.so
/usr/share/postgresql-12/extension/ltree--1.0--1.1.sql
/usr/share/postgresql-12/extension/ltree--1.1.sql
/usr/share/postgresql-12/extension/ltree.control
/usr/share/postgresql-12/extension/ltree_plpython2u--1.0.sql
/usr/share/postgresql-12/extension/ltree_plpython2u.control
/usr/share/postgresql-12/extension/ltree_plpython3u--1.0.sql
/usr/share/postgresql-12/extension/ltree_plpython3u.control
/usr/share/postgresql-12/extension/ltree_plpythonu--1.0.sql
/usr/share/postgresql-12/extension/ltree_plpythonu.control
/usr/share/postgresql-12/extension/ltree--unpackaged--1.0.sql
/usr/include/postgresql-13/server/extension/ltree/ltree.h
/usr/lib64/postgresql-13/lib64/ltree_plpython3.so
/usr/lib64/postgresql-13/lib64/ltree.so
/usr/share/postgresql-13/extension/ltree--1.0--1.1.sql
/usr/share/postgresql-13/extension/ltree--1.1--1.2.sql
/usr/share/postgresql-13/extension/ltree--1.1.sql
/usr/share/postgresql-13/extension/ltree.control
/usr/share/postgresql-13/extension/ltree_plpython2u--1.0.sql
/usr/share/postgresql-13/extension/ltree_plpython2u.control
/usr/share/postgresql-13/extension/ltree_plpython3u--1.0.sql
/usr/share/postgresql-13/extension/ltree_plpython3u.control
/usr/share/postgresql-13/extension/ltree_plpythonu--1.0.sql
/usr/share/postgresql-13/extension/ltree_plpythonu.control
```

### repoman warnings

These are not necessarily introduced with this patchset and none of them are critical however it makes sense to fix them whilst we're at it to reduce the noise:

```
RepoMan scours the neighborhood...
  ebuild.minorsyn               35
   dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 93
   dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 94
   dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 96
   dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 97
   dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 98
   dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 120
   dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 121
   dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 122
   dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 123
   dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 94
   dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 95
   dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 97
   dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 98
   dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 99
   dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 121
   dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 122
   dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 123
   dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 124
   dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 98
   dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 99
   dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 101
   dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 102
   dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 103
   dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 122
   dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 123
   dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 124
   dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 125
   dev-db/postgresql/postgresql-12.6-r1.ebuild: Ebuild contains leading spaces on line: 114
   dev-db/postgresql/postgresql-12.6-r1.ebuild: Ebuild contains leading spaces on line: 115
   dev-db/postgresql/postgresql-12.6-r1.ebuild: Ebuild contains leading spaces on line: 116
   dev-db/postgresql/postgresql-12.6-r1.ebuild: Ebuild contains leading spaces on line: 117
   dev-db/postgresql/postgresql-13.2-r1.ebuild: Ebuild contains leading spaces on line: 114
   dev-db/postgresql/postgresql-13.2-r1.ebuild: Ebuild contains leading spaces on line: 115
   dev-db/postgresql/postgresql-13.2-r1.ebuild: Ebuild contains leading spaces on line: 116
   dev-db/postgresql/postgresql-13.2-r1.ebuild: Ebuild contains leading spaces on line: 117
  metadata.warning              1
   dev-db/postgresql/metadata.xml: inconsistent use of tabs and spaces in indentation
```

**Comment by xmhd** [comment]
> Tested locally on a ~amd64 Gentoo Linux box with various USE flag combinations, no build failures. However I noticed some minor issues which are listed below:
> ### ltree
> 
> The ltree patch does not completely disable the installation of ltree related files on 9.6, 11, 12 and 13. On 9.6 the effect is minor as it only pertains to the ltree python module. The patch for version 10 works correctly. See the list of installed files below for the whole picture. Relevant flags are `USE=-ltree python`:
> 
> ```
> % qlist postgresql | grep ltree | sort -t- -k2n
> /usr/lib64/postgresql-9.6/lib64/ltree_plpython3.so
> /usr/share/postgresql-9.6/extension/ltree_plpython2u--1.0.sql
> /usr/share/postgresql-9.6/extension/ltree_plpython2u.control
> /usr/share/postgresql-9.6/extension/ltree_plpython3u--1.0.sql
> /usr/share/postgresql-9.6/extension/ltree_plpython3u.control
> /usr/share/postgresql-9.6/extension/ltree_plpythonu--1.0.sql
> /usr/share/postgresql-9.6/extension/ltree_plpythonu.control
> /usr/include/postgresql-11/server/extension/ltree/ltree.h
> /usr/lib64/postgresql-11/lib64/ltree_plpython3.so
> /usr/lib64/postgresql-11/lib64/ltree.so
> /usr/share/postgresql-11/extension/ltree--1.0--1.1.sql
> /usr/share/postgresql-11/extension/ltree--1.1.sql
> /usr/share/postgresql-11/extension/ltree.control
> /usr/share/postgresql-11/extension/ltree_plpython2u--1.0.sql
> /usr/share/postgresql-11/extension/ltree_plpython2u.control
> /usr/share/postgresql-11/extension/ltree_plpython3u--1.0.sql
> /usr/share/postgresql-11/extension/ltree_plpython3u.control
> /usr/share/postgresql-11/extension/ltree_plpythonu--1.0.sql
> /usr/share/postgresql-11/extension/ltree_plpythonu.control
> /usr/share/postgresql-11/extension/ltree--unpackaged--1.0.sql
> /usr/include/postgresql-12/server/extension/ltree/ltree.h
> /usr/lib64/postgresql-12/lib64/ltree_plpython3.so
> /usr/lib64/postgresql-12/lib64/ltree.so
> /usr/share/postgresql-12/extension/ltree--1.0--1.1.sql
> /usr/share/postgresql-12/extension/ltree--1.1.sql
> /usr/share/postgresql-12/extension/ltree.control
> /usr/share/postgresql-12/extension/ltree_plpython2u--1.0.sql
> /usr/share/postgresql-12/extension/ltree_plpython2u.control
> /usr/share/postgresql-12/extension/ltree_plpython3u--1.0.sql
> /usr/share/postgresql-12/extension/ltree_plpython3u.control
> /usr/share/postgresql-12/extension/ltree_plpythonu--1.0.sql
> /usr/share/postgresql-12/extension/ltree_plpythonu.control
> /usr/share/postgresql-12/extension/ltree--unpackaged--1.0.sql
> /usr/include/postgresql-13/server/extension/ltree/ltree.h
> /usr/lib64/postgresql-13/lib64/ltree_plpython3.so
> /usr/lib64/postgresql-13/lib64/ltree.so
> /usr/share/postgresql-13/extension/ltree--1.0--1.1.sql
> /usr/share/postgresql-13/extension/ltree--1.1--1.2.sql
> /usr/share/postgresql-13/extension/ltree--1.1.sql
> /usr/share/postgresql-13/extension/ltree.control
> /usr/share/postgresql-13/extension/ltree_plpython2u--1.0.sql
> /usr/share/postgresql-13/extension/ltree_plpython2u.control
> /usr/share/postgresql-13/extension/ltree_plpython3u--1.0.sql
> /usr/share/postgresql-13/extension/ltree_plpython3u.control
> /usr/share/postgresql-13/extension/ltree_plpythonu--1.0.sql
> /usr/share/postgresql-13/extension/ltree_plpythonu.control
> ```
> 
> ### repoman warnings
> 
> These are not necessarily introduced with this patchset and none of them are critical however it makes sense to fix them whilst we're at it to reduce the noise:
> 
> ```
> RepoMan scours the neighborhood...
>   ebuild.minorsyn               35
>    dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 93
>    dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 94
>    dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 96
>    dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 97
>    dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 98
>    dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 120
>    dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 121
>    dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 122
>    dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 123
>    dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 94
>    dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 95
>    dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 97
>    dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 98
>    dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 99
>    dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 121
>    dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 122
>    dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 123
>    dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 124
>    dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 98
>    dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 99
>    dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 101
>    dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 102
>    dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 103
>    dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 122
>    dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 123
>    dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 124
>    dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 125
>    dev-db/postgresql/postgresql-12.6-r1.ebuild: Ebuild contains leading spaces on line: 114
>    dev-db/postgresql/postgresql-12.6-r1.ebuild: Ebuild contains leading spaces on line: 115
>    dev-db/postgresql/postgresql-12.6-r1.ebuild: Ebuild contains leading spaces on line: 116
>    dev-db/postgresql/postgresql-12.6-r1.ebuild: Ebuild contains leading spaces on line: 117
>    dev-db/postgresql/postgresql-13.2-r1.ebuild: Ebuild contains leading spaces on line: 114
>    dev-db/postgresql/postgresql-13.2-r1.ebuild: Ebuild contains leading spaces on line: 115
>    dev-db/postgresql/postgresql-13.2-r1.ebuild: Ebuild contains leading spaces on line: 116
>    dev-db/postgresql/postgresql-13.2-r1.ebuild: Ebuild contains leading spaces on line: 117
>   metadata.warning              1
>    dev-db/postgresql/metadata.xml: inconsistent use of tabs and spaces in indentation
> ```

I've fixed the USE="ltree python" issue with 9.6, but I've confirmed that 11 from ::adjust has the same issue as you mentioned, so the patch needs fixed up. Neither 12 or 13 in ::adjust currently have a ltree patch all. Is this intentional? or should we be patching those two versions as well?

**Comment by xmhd** [comment]
> Tested locally on a ~amd64 Gentoo Linux box with various USE flag combinations, no build failures. However I noticed some minor issues which are listed below:
> ### ltree
> 
> The ltree patch does not completely disable the installation of ltree related files on 9.6, 11, 12 and 13. On 9.6 the effect is minor as it only pertains to the ltree python module. The patch for version 10 works correctly. See the list of installed files below for the whole picture. Relevant flags are `USE=-ltree python`:
> 
> ```
> % qlist postgresql | grep ltree | sort -t- -k2n
> /usr/lib64/postgresql-9.6/lib64/ltree_plpython3.so
> /usr/share/postgresql-9.6/extension/ltree_plpython2u--1.0.sql
> /usr/share/postgresql-9.6/extension/ltree_plpython2u.control
> /usr/share/postgresql-9.6/extension/ltree_plpython3u--1.0.sql
> /usr/share/postgresql-9.6/extension/ltree_plpython3u.control
> /usr/share/postgresql-9.6/extension/ltree_plpythonu--1.0.sql
> /usr/share/postgresql-9.6/extension/ltree_plpythonu.control
> /usr/include/postgresql-11/server/extension/ltree/ltree.h
> /usr/lib64/postgresql-11/lib64/ltree_plpython3.so
> /usr/lib64/postgresql-11/lib64/ltree.so
> /usr/share/postgresql-11/extension/ltree--1.0--1.1.sql
> /usr/share/postgresql-11/extension/ltree--1.1.sql
> /usr/share/postgresql-11/extension/ltree.control
> /usr/share/postgresql-11/extension/ltree_plpython2u--1.0.sql
> /usr/share/postgresql-11/extension/ltree_plpython2u.control
> /usr/share/postgresql-11/extension/ltree_plpython3u--1.0.sql
> /usr/share/postgresql-11/extension/ltree_plpython3u.control
> /usr/share/postgresql-11/extension/ltree_plpythonu--1.0.sql
> /usr/share/postgresql-11/extension/ltree_plpythonu.control
> /usr/share/postgresql-11/extension/ltree--unpackaged--1.0.sql
> /usr/include/postgresql-12/server/extension/ltree/ltree.h
> /usr/lib64/postgresql-12/lib64/ltree_plpython3.so
> /usr/lib64/postgresql-12/lib64/ltree.so
> /usr/share/postgresql-12/extension/ltree--1.0--1.1.sql
> /usr/share/postgresql-12/extension/ltree--1.1.sql
> /usr/share/postgresql-12/extension/ltree.control
> /usr/share/postgresql-12/extension/ltree_plpython2u--1.0.sql
> /usr/share/postgresql-12/extension/ltree_plpython2u.control
> /usr/share/postgresql-12/extension/ltree_plpython3u--1.0.sql
> /usr/share/postgresql-12/extension/ltree_plpython3u.control
> /usr/share/postgresql-12/extension/ltree_plpythonu--1.0.sql
> /usr/share/postgresql-12/extension/ltree_plpythonu.control
> /usr/share/postgresql-12/extension/ltree--unpackaged--1.0.sql
> /usr/include/postgresql-13/server/extension/ltree/ltree.h
> /usr/lib64/postgresql-13/lib64/ltree_plpython3.so
> /usr/lib64/postgresql-13/lib64/ltree.so
> /usr/share/postgresql-13/extension/ltree--1.0--1.1.sql
> /usr/share/postgresql-13/extension/ltree--1.1--1.2.sql
> /usr/share/postgresql-13/extension/ltree--1.1.sql
> /usr/share/postgresql-13/extension/ltree.control
> /usr/share/postgresql-13/extension/ltree_plpython2u--1.0.sql
> /usr/share/postgresql-13/extension/ltree_plpython2u.control
> /usr/share/postgresql-13/extension/ltree_plpython3u--1.0.sql
> /usr/share/postgresql-13/extension/ltree_plpython3u.control
> /usr/share/postgresql-13/extension/ltree_plpythonu--1.0.sql
> /usr/share/postgresql-13/extension/ltree_plpythonu.control
> ```
> 
> ### repoman warnings
> 
> These are not necessarily introduced with this patchset and none of them are critical however it makes sense to fix them whilst we're at it to reduce the noise:
> 
> ```
> RepoMan scours the neighborhood...
>   ebuild.minorsyn               35
>    dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 93
>    dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 94
>    dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 96
>    dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 97
>    dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 98
>    dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 120
>    dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 121
>    dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 122
>    dev-db/postgresql/postgresql-9.6.21-r1.ebuild: Ebuild contains leading spaces on line: 123
>    dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 94
>    dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 95
>    dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 97
>    dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 98
>    dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 99
>    dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 121
>    dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 122
>    dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 123
>    dev-db/postgresql/postgresql-10.16-r1.ebuild: Ebuild contains leading spaces on line: 124
>    dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 98
>    dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 99
>    dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 101
>    dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 102
>    dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 103
>    dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 122
>    dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 123
>    dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 124
>    dev-db/postgresql/postgresql-11.11-r1.ebuild: Ebuild contains leading spaces on line: 125
>    dev-db/postgresql/postgresql-12.6-r1.ebuild: Ebuild contains leading spaces on line: 114
>    dev-db/postgresql/postgresql-12.6-r1.ebuild: Ebuild contains leading spaces on line: 115
>    dev-db/postgresql/postgresql-12.6-r1.ebuild: Ebuild contains leading spaces on line: 116
>    dev-db/postgresql/postgresql-12.6-r1.ebuild: Ebuild contains leading spaces on line: 117
>    dev-db/postgresql/postgresql-13.2-r1.ebuild: Ebuild contains leading spaces on line: 114
>    dev-db/postgresql/postgresql-13.2-r1.ebuild: Ebuild contains leading spaces on line: 115
>    dev-db/postgresql/postgresql-13.2-r1.ebuild: Ebuild contains leading spaces on line: 116
>    dev-db/postgresql/postgresql-13.2-r1.ebuild: Ebuild contains leading spaces on line: 117
>   metadata.warning              1
>    dev-db/postgresql/metadata.xml: inconsistent use of tabs and spaces in indentation
> ```

Ok so:

- fixed the 9.6 issue
- current ebuild for 11 in ::adjust applies the patch but does not have a USE flag, nor $(use_with ltree) in src_configure. I have added
- current ebuilds for 12 and 13 in ::adjust do not apply the patch nor have a USE flag or $(use_with ltree) in src_configure.

I have fixed 11 by adding the USE flag and $(use_with ltree), and have copied the same logic into 12 and 13. Confirmed working locally on my ~amd64 machine.

My next question however: why do the current 12 + 13 ebuilds in ::adjust not have this logic, and why does 11 only have it half implemented? Is this by design? or it just slipped under the radar?

**Comment by alip** [comment]
> My next question however: why do the current 12 + 13 ebuilds in ::adjust not have this logic, and why does 11 only have it half implemented? Is this by design? or it just slipped under the radar?

The ltree USE flag and the respective patch may be dropped. We no longer need that. The historical rationale was the fact that we had a fork of the upstream ltree and we wanted to prevent clashes by disabling the build of the in-tree ltree extension. This extension has been renamed to [adjust/wltree](https://github.com/adjust/wltree) so there're no potential clashes anymore. @einhverfr confirmed this.

About cassert USE flag which is present for 9.6, we want to add this USE flag to recent versions of postgresql as well and restart using it to test our extensions. This is required to detect some memory bugs which are hard to reproduce on production systems. See:

https://github.com/postgres/postgres/blob/bcf2667bf62d72faced64cb60ffbd2e599a0ebe8/configure.ac#L692-L697

which confirms we can do this identically as we do it for the 9.6 ebuild.

@xmhd, would you be so kind to clean up the ltree use flag and patch from all ebuilds and add cassert to all the recent ebuilds? TIA.

**Comment by xmhd** [comment]
> > My next question however: why do the current 12 + 13 ebuilds in ::adjust not have this logic, and why does 11 only have it half implemented? Is this by design? or it just slipped under the radar?
> 
> The ltree USE flag and the respective patch may be dropped. We no longer need that. The historical rationale was the fact that we had a fork of the upstream ltree and we wanted to prevent clashes by disabling the build of the in-tree ltree extension. This extension has been renamed to [adjust/wltree](https://github.com/adjust/wltree) so there're no potential clashes anymore. @einhverfr confirmed this.
> 
> About cassert USE flag which is present for 9.6, we want to add this USE flag to recent versions of postgresql as well and restart using it to test our extensions. This is required to detect some memory bugs which are hard to reproduce on production systems. See:
> 
> https://github.com/postgres/postgres/blob/bcf2667bf62d72faced64cb60ffbd2e599a0ebe8/configure.ac#L692-L697
> 
> which confirms we can do this identically as we do it for the 9.6 ebuild.
> 
> @xmhd, would you be so kind to clean up the ltree use flag and patch from all ebuilds and add cassert to all the recent ebuilds? TIA.

Ack. Ebuilds have been updated to remove ltree USE and patch. Cassert USE flag has been added to the others.

So now our only difference from ::gentoo is the cassert USE flag. I plan on submitting a pull request to Gentoo so that we can unfork Postgres from our overlay. Gentoo maintains Postgres v9.5 -> v13, so there should be no issues for us there.

**Review by alip — APPROVED** [review]

**Pull request merged**